src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -83,7 +83,8 @@ TaskRequest ResourceMapToTaskRequest(
     } else if (resource.first == ray::kMemory_ResourceLabel) {
       task_request.predefined_resources[MEM].demand = resource.second;
     } else {
-      task_request.custom_resources[i].id = string_to_int_map.Insert(resource.first);
+      string_to_int_map.Insert(resource.first);
+      task_request.custom_resources[i].id = string_to_int_map.Get(resource.first);
```

Comment:
This was a latent bug. `string_to_int_map.Insert` only returns the ID if it already existed. Therefore, if this `Insert` call is the first call, everything breaks :(

Reply:
Will there be any issue if we make it return ID always?

Reply:
I think it's non trivial because StringIdMap takes an option argument `num_ids`